src/core/net/srp_client.cpp

```diff
@@ -1418,7 +1456,11 @@ Error Client::AppendSignature(Message &aMessage, Info &aInfo)
     sha256.Update(aMessage, 0, offset);
 
     sha256.Finish(hash);
+#if OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE
+    SuccessOrExit(error = aInfo.mKeyRef.Sign(hash, signature));
```

Comment:
nit: Some of these `#if`s could be avoided if `mKeyRef` was renamed to `mKeyPair` (which is OK since both are mutually exclusive).

Reply:
That is true, but I opted this way deliberately to let the user know that we are dealing with keyrefs here.

Reply:
oh, I see.